pytripgui/canvas_vc/canvas_controller.py

```diff
@@ -90,10 +92,38 @@ def set_patient(self, patient, state):
             self._ui.voi_list.event_callback = self._on_update_voi
             self._ui.voi_list.fill(patient.vdx.vois, lambda item: item.name)
 
-        self._ui.set_position_changed_callback(self.set_current_slice_no)
         self.update_canvas_view()
+
+        self._update_slider_values(position_safe=True, perspective_safe=True)
+
         self._ui.draw()
 
+    def _update_slider_values(self, position_safe: bool, perspective_safe: bool):
+        """
+        Updates min, max and current values of ui slider.
+        Can be "safe" position-callback-wise and perspective-callback-wise,
+            which means that callback will not be invoked id proper flag is set
+
+        :param bool position_safe: if set, position change callback will be suppressed
+        :param bool perspective_safe: if set, perspective change callback will be suppressed
+        """
+        # remove event listeners
+        if position_safe:
+            self._ui.remove_position_changed_callback(self.set_current_slice_no)
+        if perspective_safe:
+            self._ui.internal_events.on_perspective_change -= self._perspective_has_changed_callback
+
+        # set parameters
```

Comment:
Why these parameters (`max_position` and `perspective`) are added here ? 
They should be independent of the slider position and should not be changed when user moves slider

Reply:
They are not dependent! And they need to be changed when perspective or patient has changed - and they are.
It is a method that loads stored information - how many slices in patient/perspective there are and where user stopped scrolling (current position)

Reply:
Then the docstring should mention that. I see now that `_update_slider_values` is being also called on changing patient/perspective. This is not obvious and should be mentioned in the docstring.

Reply:
Well, `_update_slider_values` is being called - not also, but only - on changing patient or perspective,
![image](https://user-images.githubusercontent.com/44035398/131675037-3bd478ae-6e9c-4c22-9d02-2d6e1c7f1a40.png)
![image](https://user-images.githubusercontent.com/44035398/131675050-41f426fb-774c-4258-94b9-dd047da5e1e8.png)
![image](https://user-images.githubusercontent.com/44035398/131675059-6ff16dea-46ce-4e07-8703-748c6c095c1b.png)

In my opinion, docstring should not say where method is being called, but what it does and why it is called should be commented in line before call, if it is not obvious.

I mean, I do not understand what I have to do

Reply:
I changed it to be more precise
![image](https://user-images.githubusercontent.com/44035398/131683230-5c291ac1-0b75-4479-8327-9fca4bd66dcb.png)

Reply:
Now its as expected !